DuckDuckGo/Subscription/ViewModel/SubscriptionContainerViewModel.swift

```diff
@@ -31,11 +32,18 @@ final class SubscriptionContainerViewModel: ObservableObject {
     let email: SubscriptionEmailViewModel
     
     
-    init(userScript: SubscriptionPagesUserScript = SubscriptionPagesUserScript(),
-         subFeature: SubscriptionPagesUseSubscriptionFeature = SubscriptionPagesUseSubscriptionFeature()) {
+    init(
+        origin: String?,
+        userScript: SubscriptionPagesUserScript,
+        subFeature: SubscriptionPagesUseSubscriptionFeature
+    ) {
+        var purchaseURL = URL.subscriptionPurchase
+        if let origin {
+            purchaseURL = purchaseURL.appendingParameter(name: AttributionParameter.origin, value: origin)
+        }
```

Comment:
I'd probably avoid handling creation of `purchaseURL` here. Instead just take the `origin`  and and pass it into `SubscriptionFlowViewModel`'s init (instead of `purchaseURL`)

Reply:
Done!